tests/sqlparser_common.rs

```diff
@@ -854,8 +854,8 @@ fn parse_cast() {
     );
 
     one_statement_parses_to(
-        "SELECT CAST(id AS decimal) FROM customer",
-        "SELECT CAST(id AS numeric) FROM customer",
+        "SELECT CAST(CELL(a) AS decimal) FROM customer",
```

Comment:
This was supposed to test the parsing of `decimal` as `numeric`. The fact that more complex expressions can be nested in CAST can be tested separately.